BlueprintUI/Sources/Layout/Stack.swift

```diff
@@ -448,21 +454,88 @@ extension StackLayout {
         for items: [(traits: Traits, content: Measurable)],
         in vectorConstraint: VectorConstraint
     ) -> [Segment] {
-        let constraint = vectorConstraint.constraint(axis: axis)
 
-        /// The measured sizes of each item, constrained as if each were the only element in the stack.
-        let basisSizes = items.map { $0.content.measure(in: constraint).axis(on: axis) }
+        let minimumTotalSpacing = CGFloat(items.count - 1) * minimumSpacing
+
+        func basisSizes() -> [CGFloat] {
+            let constraint = vectorConstraint.constraint(axis: axis)
+
+            if BlueprintStackUpdates.isOptimizedLayoutEnabled {
+                final class Measurement {
+
+                    let item: (traits: Traits, content: Measurable)
+                    var size: CGSize?
+
+                    let isFixed: Bool
+
+                    init(item: (traits: StackLayout.Traits, content: Measurable)) {
+                        self.item = item
+                        size = nil
+
+                        isFixed = item.traits.growPriority == 0 && item.traits.shrinkPriority == 0
+                    }
+                }
+
+                let measurements: [Measurement] = items.map { item in
+                    Measurement(item: item)
+                }
+
+                // Measure fixed items
+
+                measurements.forEach { measurement in
+                    guard measurement.isFixed else { return }
+
+                    measurement.size = measurement
+                        .item
+                        .content
+                        .measure(in: constraint)
+                }
+
+                // Determine how much space the fixed items took up
+
+                let fixedMagnitude: CGFloat = measurements.reduce(0) { magnitude, measurement in
+                    magnitude + (measurement.size?.axis(on: axis) ?? 0)
+                }
+
+                // From that, determine how much space the flexible items will have
+
+                let flexibleMagnitude: SizeConstraint.Axis = constraint.axis(on: axis) - fixedMagnitude - minimumTotalSpacing
+
+                // TODO: What if flexibleMagnitude is negative??
```

Comment:
Looking for input on what to do here – eg if there's one or more fixed elements that take up all the space. I was thinking we may just want to skip measuring the flexible elements entirely, since they basically can't lay out.

Reply:
I might be misunderstanding the issue, but I'd expect we could skip measurement of those flexible items and just use a size of `0`, right?

Reply:
I think we could – let me see what happened before at layout time.

Reply:
I added a test (`test_fixed_elements_overflow_behavior`), and the behavior here mirrors the old behavior – which seems wrong; but at least it's consistent. I'll follow up with lower bounding this to zero.

Reply:
https://github.com/square/Blueprint/pull/363